gpt_json/gpt.py

```diff
@@ -144,6 +139,58 @@ async def run(
         else:
             return self.schema_model(**extracted_json), fixed_payload
 
+    async def stream(
+        self,
+        messages: list[GPTMessage],
+        max_response_tokens: int | None = None,
+        format_variables: dict[str, Any] | None = None,
+    ) -> AsyncGenerator[StreamingObject[SchemaType], None]:
+        """
+        See `run` for documentation. This method is an async generator wrapper around `run` that streams partial results
+        instead of returning them all at once.
+
+        :return: yields `StreamingObject[SchemaType]`s. 
+        """
+        if self.extract_type != ResponseType.DICTIONARY:
+            raise NotImplementedError("For now, streaming is only supported for dictionary responses.")
+        for field_type in self.schema_model.__annotations__.values():
+            if field_type != str:
+                raise NotImplementedError("For now, streaming is only supported for dictionary responses with string fields.")
+        
+        messages = [
+            self.fill_message_template(message, format_variables or {})
+            for message in messages
+        ]
+
+        backoff_request_submission = await self._backoff_submit_request()
+
+        raw_responses = await backoff_request_submission(messages, max_response_tokens=max_response_tokens, stream=True)
+
+        prev_partial = None
+        cumulative_response = ""
+        async for raw_response in raw_responses:
+            logger.debug(f"------- RAW RESPONSE ----------")
+            logger.debug(raw_response)
+            logger.debug(f"------- END RAW RESPONSE ----------")
+
+            response = ChatCompletionChunk.from_dict(raw_response)
+
+            if response.choices[0].delta.role is not None:
+                # Ignore assistant role message
+                continue
+            if response.choices[0].finish_reason is not None:
+                # Ignore finish message 
+                continue
+            
+            cumulative_response += response.choices[0].delta.content
+            
+            partial_data, event = parse_streamed_json(cumulative_response)
+            partial_response = self.streaming_type(partial_data, prev_partial, event)
```

Comment:
Why are you adding this as an instance method versus just dynamically creating the `StreamingObject` at the start of this function? Is there something with the typehinting that was not validating if you try to create inline?